lib/mapObjects/CQuest.h

```diff
@@ -15,13 +15,25 @@
 #include "../CCreatureSet.h"
 #include "../NetPacksBase.h"
 
+namespace std
+{
+	// Used in std::unordered_map
+	template<> struct hash<VCMI_LIB_WRAP_NAMESPACE(ArtifactID)>
+	{
+		size_t operator()(const VCMI_LIB_WRAP_NAMESPACE(ArtifactID) & aid) const
+		{
+			return hash<int>{}(aid.num);
+		}
+	};
+}
```

Comment:
Extending std is usually not what you are supposed to do.
Instead of having hash function in std namespace, make separate functor and pass it to unordered_map as template argument (for me it seems like improvement desired)

Reply:
Actually providing template specializations in std namespace is one of explicitly allowed exceptions in the standard.
So this code is completely legal and won't cause any issues with any standard-compliant c++ compiler.

Extending std namespace vs providing functor to unordered_map is more of a code style question.
And current approach allows declaring unordered_map's seamlessly without any additional code.

Reply:
> Opening namespace std can accidentally introduce undefined behavior, and is not necessary for specializing class templates.

From: https://en.cppreference.com/w/cpp/language/extending_std
I've put a suggestion